# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../gkrellm/libsm.patch
# Copyright (C) 2010 The OpenSDE Project
# Copyright (C) 2006 The T2 SDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

diff -ruN gkrellm-2.3.4-orig/Makefile gkrellm-2.3.4/Makefile
--- gkrellm-2.3.4-orig/Makefile	2010-01-05 00:13:13.000000000 +0100
+++ gkrellm-2.3.4/Makefile	2010-08-25 19:00:46.800005755 +0200
@@ -247,7 +247,7 @@
 	(cd po && ${MAKE} all)
 	(cd src && ${MAKE} \
 		EXTRAOBJS= SYS_LIBS="-lkvm" \
-		SMC_LIBS="-L/usr/X11R6/lib -lSM -lICE -Wl,-R/usr/X11R6/lib" \
+		SMC_LIBS="`pkg-config --libs sm`" \
 		gkrellm )
 	(cd server && ${MAKE} \
 		EXTRAOBJS= SYS_LIBS="-lkvm" gkrellmd )
@@ -256,7 +256,7 @@
 	(cd po && ${MAKE} all)
 	(cd src && ${MAKE} \
 		EXTRAOBJS= SYS_LIBS="-lkvm -pthread" \
-		SMC_LIBS="-L/usr/X11R6/lib -lSM -lICE -R/usr/X11R6/lib" \
+		SMC_LIBS="`pkg-config --libs sm`" \
 		gkrellm )
 	(cd server && ${MAKE} \
 		EXTRAOBJS= SYS_LIBS="-lkvm -pthread" gkrellmd )
diff -ruN gkrellm-2.3.4-orig/src/Makefile gkrellm-2.3.4/src/Makefile
--- gkrellm-2.3.4-orig/src/Makefile	2010-01-05 00:12:07.000000000 +0100
+++ gkrellm-2.3.4/src/Makefile	2010-08-25 19:00:46.800005755 +0200
@@ -20,7 +20,7 @@
 MANDIRMODE ?= 755
 INSTALL ?= install
 LINK_FLAGS ?= -Wl,-E
-SMC_LIBS ?= -L/usr/X11R6/lib -lSM -lICE
+SMC_LIBS ?= $(shell pkg-config --libs sm)
 
 X11_LIBS ?= -lX11
 
@@ -188,11 +188,11 @@
 
 netbsd1:
 	$(MAKE) EXTRAOBJS= SYS_LIBS="-lkvm" \
-		SMC_LIBS="-L/usr/X11R6/lib -lSM -lICE -Wl,-R/usr/X11R6/lib" \
+		SMC_LIBS="`pkg-config --libs sm`" \
 		gkrellm
 netbsd2:
 	$(MAKE) EXTRAOBJS= SYS_LIBS="-lkvm -pthread" \
-		SMC_LIBS="-L/usr/X11R6/lib -lSM -lICE -R/usr/X11R6/lib" \
+		SMC_LIBS="`pkg-config --libs sm`" \
 		gkrellm
 
 openbsd:
